lib/gems/pending/util/vmdb-logger.rb

```diff
@@ -108,9 +108,13 @@ def self.log_hashes(logger, h, options = {})
     filter = Array(options[:filter]).flatten.compact.map(&:to_s) << "password"
     filter.uniq!
 
-    YAML.dump(h).split("\n").each do |l|
+    values = YAML.dump(h).gsub(MiqPassword::REGEXP) { |m| m.gsub($2, "FILTERED") }
+    values.split("\n").each do |l|
       next if l[0...3] == '---'
-      logger.send(level, "  #{l}") unless filter.any? { |f| l.include?(f) }
+      if key = filter.detect { |f| l.include?(f) }
+        l.gsub!(/#{key}.*: (.+)/) { |m| m.gsub!($1, "[FILTERED]") }
```

Comment:
The method was doing a YAML.dump before which means we do not have to be concerned with the depth of embedded objects.

If we stick with the YAML string I think we could do a `gsub!` for any filters over the whole string outside of the  `.each` loop.  Also, would `.each_line` work here instead of `.split("\n")`?